src/display/node_stream.js

```diff
@@ -0,0 +1,374 @@
+/* Copyright 2012 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* globals __non_webpack_require__ */
+
+let fs = __non_webpack_require__('fs');
+let http = __non_webpack_require__('http');
+let https = __non_webpack_require__('https');
+let url = __non_webpack_require__('url');
+
+import { assert, createPromiseCapability } from '../shared/util';
+import { validateRangeRequestCapabilities } from './network_utils';
+
+class PDFNodeStream {
+  constructor(options) {
+    this.options = options;
+    this.source = options.source;
+    this.url = url.parse(this.source.url);
+    this.isHttp = this.url.protocol === 'http:' ||
+                  this.url.protocol === 'https:';
+    this.isFsUrl = !this.url.host;
```

Comment:
> What are the expected values for url that you intend to support here?

I am checking here if url refers to the filesystem or not, so that we can choose between `PDFNodeStream***Reader` or `PDFNodeStreamFs***Reader`.

I am trying to parse URL using `url.parse`, and then checking for host. In most of cases I found that `url`s are of the form `../....path/to/some/pdf.pdf` or if `url` is absolute `home/userName/path/to/some/pdf.pdf` or `file:///home/user/path/to/some/pdf.pdf`-- (and all these cases works fine with above check). The only case that I found failing is `file://localhost/path/to/some/pdf.pdf`, and I think this can be fix by checking for `protocol === file:`.

Is it sounds reasonable to you, or you have something else in your mind? If not, can you suggest me something concrete?